app/Http/Controllers/Admin/BeneficiaryController.php

```diff
@@ -193,118 +195,230 @@ public function update_call_champion_report($due_list){
+* @algorithm
+* 1. Load Excel file and data will be stored in $reader (array).
+* 2. Apply foreach on this array and get data of each row of excel file.
+* 3. On each data apply validation rules and accordingly store the message for 		particular type of error
+*/	
 
-}
\ No newline at end of file
+	public function importExcel()
+	{
+		//this variable tells whether data is error free and can be stored or not.
+		Session::flash('should_data_be_inserted',1);
+		//variable for different types of errors and warnings.
+		Session::flash('count_excelupload_errors.count',0);
+		Session::flash('count_excelupload_warning.count',0);
+		Session::flash('count_excelupload_data_repeated.count',0);	
+		//loads excel file
+		Excel::load(Input::file('beneficiaries_data'),function($reader)
+ 		{
+
+ 			//$reader acts differently in different excel formats
+			if(Input::file('beneficiaries_data')->getClientOriginalExtension()=='csv')
+				$data=$reader->get();
+			else
+			{
+				$data=$reader->get();
+				$data=$data[0];
+			}
+			  $data->each(function($r){
+					$beneficiary= new Beneficiary;
+					//validation rules
+		  			$rules = array(
+		  			 'v_name' => 'required|min:3|max:20|Regex:/^[ A-Za-z]+[A-Za-z0-9.\' ]*$/',
+  					'v_phone_number'=>'required|numeric|digits_between:10,10',
+  					'dt_due_date'=>'required'
+		  			);
+ 				$dt_due_date=str_replace("-", "/", $r->date_of_delivery);
+    			$beneficiary_data = array(
+    					'fk_f_id'=>$r->field_worker_id,
+    					'v_name' => $r->womans_name,
+    					'v_husband_name'=>$r->fatherspouse_name,
+    					'i_age'=>$r->age,
+    					'v_phone_number'=>$r->mobile_no,
+    					'v_awc_number'=>$r->awc_code,
+    					'v_village_name'=>$r->village_name,
+    					'dt_due_date'=>''//date('d/m/y',strtotime($dt_due_date))
+    			);
+    			//validating dates as by default date value is 1970/1/1 thus chaning that to null so that validation rules work correctly
+    			if($r->date_of_delivery!='')
+    			{
+    				$beneficiary_data['dt_due_date']=date('d/m/y',strtotime($dt_due_date));
+    				$var=Carbon::createFromFormat('d/m/y', $beneficiary_data['dt_due_date']);
+    				if($var!='')
+    				$beneficiary_data['dt_due_date']=$var;
+    				else
+					$beneficiary_data['dt_due_date']=null;
+    			}
+    			else
+    			{
+    				$beneficiary_data['dt_due_date']=null;	
+    			}
+    			// checks for multiple combinations of mother name and hsuband name in database.
+ 				$already_exist_name=Beneficiary::where(['v_name'=>$beneficiary_data['v_name'],'v_husband_name'=>$beneficiary_data['v_husband_name']])->first();
+ 				if($already_exist_name['v_name']!='')
+ 				{
+    				$count_excelupload_warning=Session::get('count_excelupload_warning.count');
+ 					Session::flash('count_excelupload_warning'.$count_excelupload_warning,$r->srno);
+ 					Session::flash('count_excelupload_warning.message'.$count_excelupload_warning,'Combination of Mother name and Husband name already exist');					
```

Comment:
sir how to do this ??
"please put all these static string in a lang file and access it via
trans(file_name.key_name)
i would want to review all these error messages in one place"

On Tue, Aug 16, 2016 at 1:26 PM, shashank-srikant <notifications@github.com>
wrote:

> In app/Http/Controllers/Admin/BeneficiaryController.php
> <https://github.com/shashank-srikant/sevasetu-mothercare/pull/32#discussion_r74890489>
> :
>
> > +    				if($var!='')
> > +    				$beneficiary_data['dt_due_date']=$var;
> > +    				else
> > +					$beneficiary_data['dt_due_date']=null;
> > +    			}
> > +    			else
> > +    			{
> > +    				$beneficiary_data['dt_due_date']=null;	
> > +    			}
> > +    			// checks for multiple combinations of mother name and hsuband name in database.
> > + 				$already_exist_name=Beneficiary::where(['v_name'=>$beneficiary_data['v_name'],'v_husband_name'=>$beneficiary_data['v_husband_name']])->first();
> > + 				if($already_exist_name['v_name']!='')
> > + 				{
> > +    				$count_excelupload_warning=Session::get('count_excelupload_warning.count');
> > + 					Session::flash('count_excelupload_warning'.$count_excelupload_warning,$r->srno);
> > + 					Session::flash('count_excelupload_warning.message'.$count_excelupload_warning,'Combination of Mother name and Husband name already exist');					
>
> @Shubham2801 <https://github.com/Shubham2801>
> please put all these static string in a lang file and access it via
> trans(file_name.key_name)
> i would want to review all these error messages in one place
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/shashank-srikant/sevasetu-mothercare/pull/32/files/7cc582f20e7bcfaf300c461b89c3b8771db6b267#r74890489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AT4BkgBM8xleH5M0N5FRhInhIbZGZsiuks5qgW0YgaJpZM4JeUVt>
> .
>